### PR TITLE
5910: VirtualKey doesn't need to be comparable

### PR DESCRIPTION
**Pull request by artemananiev**
Fix summary:

* `VirtualKey` no longer implements `Comparable`
* Since it was the only reason to have a class type for virtual keys, and not it's removed, `VirtualKey` is no longer generic
* `VirtualLongKeyTest` is removed as it was only checking long comparations

Fixes: https://github.com/hashgraph/hedera-services/issues/5910
Signed-off-by: Artem Ananev <artem.ananev@swirldslabs.com>

### CODE REVIEW

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/jasperdb/files/hashmap/VirtualKeySetSerializer.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
  * purpose key serializer.
  */
 @ConstructableIgnored
-public class VirtualKeySetSerializer implements KeySerializer<VirtualKey<VirtualLongKey>> {
+public class VirtualKeySetSerializer implements KeySerializer<VirtualKey> {
```
Comment (OlegMazurov) [inline platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/jasperdb/files/hashmap/VirtualKeySetSerializer.java:35]: Could/should it be `implements KeySerializer<VirtualLongKey>`?

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/VirtualKeySetSerializer.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
  * general purpose key serializer.
  */
 @ConstructableIgnored
-public class VirtualKeySetSerializer implements KeySerializer<VirtualKey<VirtualLongKey>> {
+public class VirtualKeySetSerializer implements KeySerializer<VirtualKey> {
```
Comment (OlegMazurov) [inline platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/VirtualKeySetSerializer.java:36]: Same as its `jasperdb` counterpart: `VirtualLongKey`

**Review by imalygin — APPROVED** [review]

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/jasperdb/files/hashmap/VirtualKeySetSerializer.java (hunk 1 above)
Comment (artemananiev) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/jasperdb/files/hashmap/VirtualKeySetSerializer.java:35]: Makes sense. Let me implement it

File: platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/VirtualKeySetSerializer.java (hunk 2 above)
Comment (artemananiev) [reply platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/hashmap/VirtualKeySetSerializer.java:36]: +1

**Review by imalygin — APPROVED** [review]

**Review by hendrikebbers — APPROVED** [review]

**Review by jsync-swirlds — APPROVED** [review]

**Review by poulok — APPROVED** [review]
Approving based on review of the 5 hashgraph team owned files

**Pull request merged**